bofire/surrogates/multi_task_gp.py

```diff
@@ -0,0 +1,105 @@
+import warnings
+from typing import Dict, Optional
+
+import botorch
+import numpy as np
+import pandas as pd
+import torch
+from botorch.fit import fit_gpytorch_mll
+from botorch.models.transforms.outcome import Standardize
+from gpytorch.mlls import ExactMarginalLogLikelihood
+
+import bofire.kernels.api as kernels
+import bofire.priors.api as priors
+from bofire.data_models.enum import OutputFilteringEnum
+from bofire.data_models.features.api import TaskInput
+from bofire.data_models.priors.api import LKJPrior
+
+# from bofire.data_models.molfeatures.api import MolFeatures
+from bofire.data_models.surrogates.api import MultiTaskGPSurrogate as DataModel
+from bofire.data_models.surrogates.scaler import ScalerEnum
+from bofire.surrogates.botorch import BotorchSurrogate
+from bofire.surrogates.trainable import TrainableSurrogate
+from bofire.surrogates.utils import get_scaler
+from bofire.utils.torch_tools import tkwargs
+
+
+class MultiTaskGPSurrogate(BotorchSurrogate, TrainableSurrogate):
+    def __init__(
+        self,
+        data_model: DataModel,
+        **kwargs,
+    ):
+        self.n_tasks = len(data_model.inputs.get(TaskInput).features[0].categories)
+        self.kernel = data_model.kernel
+        self.scaler = data_model.scaler
+        self.output_scaler = data_model.output_scaler
+        self.noise_prior = data_model.noise_prior
+        self.task_prior = data_model.task_prior
+        if isinstance(data_model.task_prior, LKJPrior):
+            # set the number of tasks in the prior
+            self.task_prior.n_tasks = self.n_tasks
+        # obtain the name of the task feature
+        self.task_feature_key = data_model.inputs.get_keys(TaskInput)[0]
+
+        super().__init__(data_model=data_model, **kwargs)
+
+    model: Optional[botorch.models.MultiTaskGP] = None
+    _output_filtering: OutputFilteringEnum = OutputFilteringEnum.ALL
+    training_specs: Dict = {}
+
+    def _fit(self, X: pd.DataFrame, Y: pd.DataFrame):
+        scaler = get_scaler(self.inputs, self.input_preprocessing_specs, self.scaler, X)
+        transformed_X = self.inputs.transform(X, self.input_preprocessing_specs)
+
+        tX, tY = torch.from_numpy(transformed_X.values).to(**tkwargs), torch.from_numpy(
+            Y.values
+        ).to(**tkwargs)
+
+        self.model = botorch.models.MultiTaskGP(  # type: ignore
+            train_X=tX,
+            train_Y=tY,
+            task_feature=X.columns.get_loc(
```

Comment:
```suggestion
            task_feature=transformed_X.columns.get_loc(
```

I think we have to take it from here in case of possible transformations with respect to onehots for categorical ones etc.